plugins/catalog-backend/src/next/Stitcher.ts

```diff
@@ -59,9 +60,22 @@ export class Stitcher {
 
   async stitch(entityRefs: Set<string>) {
     for (const entityRef of entityRefs) {
-      await this.transaction(async txOpaque => {
-        const tx = txOpaque as Knex.Transaction;
-
+      try {
+        const ticket = uuid();
+        const ticketRows = await this.database<DbFinalEntitiesRow>(
+          'final_entities',
+        )
+          .update('stitch_ticket', ticket)
+          .whereIn('entity_id', qb =>
+            qb
+              .select('entity_id')
+              .from('refresh_state')
+              .where({ entity_ref: entityRef }),
+          );
+        if (ticketRows === 0) {
+          // No ticket written.
```

Comment:
Should mention that this is because the `entityRef` referenced an entity that doesn't exist in `refresh_state`